app/Http/Controllers/ProductController.php

```diff
@@ -13,36 +14,42 @@ class ProductController extends Controller
     public function index()
     {
         $products = Product::select('id', 'name', 'brand_id', 'color_id', 'size_id', 'status')->paginate(10);
-          
+
         return view('admin.pages.product.index', compact('products'));
     }
     public function add()
     {
         $brands = Brand::select('id', 'name')->get();
         $colors = Color::select('id', 'name')->get();
-        $sizes = Size::select('id', 'name')->get();    
-        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes'));
+        $sizes = Size::select('id', 'name')->get();
+        $categories = Category::select('id', 'name')->get();
+
+        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes', 'categories'));
     }
     public function store(ProductValidation $request)
     {
-        Product::create($request->validated());
+        $product = Product::create($request->validated());
+        $product->attachCategories($request->input('category_ids'));
 
-        return redirect()->route('product.index')->with('success',  'Product added successfully ');
+        return redirect()->route('product.index')->with('success',  'Product added successfully.');
     }
     public function edit(string $productId)
     {
         $brands = Brand::select('id', 'name')->get();
         $colors = Color::select('id', 'name')->get();
-        $sizes = Size::select('id', 'name')->get(); 
+        $sizes = Size::select('id', 'name')->get();
+        $categories = Category::select('id', 'name')->get();
+
         $product = Product::select('id', 'name', 'brand_id', 'size_id', 'color_id', 'status')->find($productId);
 
-        return view('admin.pages.product.form', compact('product', 'brands', 'colors', 'sizes'));
+        return view('admin.pages.product.form', compact('product', 'brands', 'colors', 'sizes', 'categories'));
     }
 
     public function update(ProductValidation $request, Product $product)
     {
         $product->update($request->validated());
```

Comment:
now you are attaching category with product. so what if category trying to delete and that is connected with product?

Reply:
sir error ma samjatu nathi